### DISCUSSION TITLE
Multiple areas

### DISCUSSION THREAD

**Discussion by damylen** [onthegomap/planetiler#255] [Q&A] [answered]
Currently I am using the docker setup as described and specify the area (downloaded from geofabrik) as specified in the readme and use tileserver-gl for hosting the tiles. This works great! 

Is there a solution to work with multiple areas in a single config? I would prefer to work with two countries instead of downloading, processing and hosting a continent.

**Comment by wipfli** [comment, marked as answer]
Would you supply multiple input files or rather one input file plus multiple bounding boxes?

**Reply by wipfli** [reply]
What would be cool is if we could give planetiler an input file plus a polygon an get tiles which only contain features which are in the polygon.

**Reply by wipfli** [reply]
There was a question once in the chat how to get stuff which is inside airport polygons or so. @msbarry pointed out that you can hook into processFeatures and do checks there. https://github.com/onthegomap/planetiler/blob/main/planetiler-core/src/main/java/com/onthegomap/planetiler/geo/PolygonIndex.java might help to see if something is inside a polygon

**Reply by damylen** [reply]
> Would you supply multiple input files or rather one input file plus multiple bounding boxes?

Something simple like: area=france,italy would be helpful. Could I already create something similar with multiple commands?

**Reply by msbarry** [reply]
Cool! I've got a TODO in the code to handle multiple osm.pbf files, this would be a logical extension from that. It would take a bit of work, and corner cases to handle (like the same element in multiple input files) but I think it would be doable. 

Two approaches you could try in the meantime would be

1) run with area=france then again with area=italy and combine the output mbtiles using tile-join from tippecanoe 
2) or combine the input pbf files using osmium merge then run planetiler on the output with --osm-path=...
